docs/usage-as-end-user.md

```diff
@@ -96,9 +96,14 @@ serverGroupConfig:
   policy: soft-anti-affinity
 ```
 
-Currently, the only configuration available is the `serverGroupConfig`. If this option is set, then a new server group will be created with the configured policy and all machines managed by this worker pool will be assigned as members of the group.
-
-Please note that the available options for `serverGroupConfig` are defined in the provider specific section of `CloudProfile`.
+When you specify `serverGroupConfig` in your worker configuration, a new server group will be created with the configured policy and all machines managed by this worker will be assigned as members of the created server group.
+
+Please note the following restrictions when deploying workers with server groups:
++ The `serverGroupConfig` section is optional, but if it is included in the shoot spec, it must contain a valid policy value.
++ The available `policy` values that can be used, are defined in the provider specific section of `CloudProfile`, by your operator.
++ Certain policy values may induce further constraints. Using the `affinity` value is only allowed when the workers utilize a single zone.
++ The `serverGroupConfig` and `serverGroupConfig.policy` fields are immutable upon creation. Users are not allowed to change the policy value and neither add or remove a `serverGroupConfig` section from a worker after it has been created.
+Users can on the other hand, add new workers managed by server groups to an existing shoot and migrate their workloads to the new workers.
```

Comment:
OK, now reading this, the first sentence (with my comment about in-place) is clarified. Maybe this can be made more clear from the beginning, i.e., that all of this only applies for newly created worker pools while existing pools are immutable.